tfhe/src/shortint/server_key/mod.rs

```diff
@@ -385,6 +385,41 @@ impl ServerKey {
         })
     }
 
+    /// Given a function as input, constructs the lookup table working on the message bits
+    /// Carry bits are ignored
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::shortint::gen_keys;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys(PARAM_MESSAGE_2_CARRY_2_KS_PBS);
+    ///
+    /// let msg = 3;
+    ///
+    /// let ct = cks.encrypt(msg);
+    ///
+    /// // Generate the lookup table for the function f: x -> x^2 mod 2^2
+    /// let f = |x| x ^ 2;
+    ///
+    /// let acc = sks.generate_msg_lookup_table(f, ct.message_modulus);
+    /// let ct_res = sks.apply_lookup_table(&ct, &acc);
+    ///
+    /// let dec = cks.decrypt(&ct_res);
+    /// // 3^2 mod 4 = 1
+    /// assert_eq!(dec, f(msg));
```

Comment:
x^2 in the lut means xor(x, 2) not power so this might confuse readers

Reply:
Yes, I thought it was a pow :upside_down_face:

Reply:
Should I change that to `x * x` ?

Reply:
the corresponding PR from Thomas has been accepted so should get merged fairly soon, so if you rebase and update should be ok 👍

Reply:
Done